### ISSUE TITLE
Loopless solution not working with glpk

### ISSUE THREAD

**Issue by sumanasri** [opencobra/cobrapy#877]
HI, I tried getting a loopless solution with CycleFreeFlux approach but it does not remove the loop from the flux distributuion. I tried the Schellenber et. al. “classical” loopless formulation but it exits without doing anything when I use it with glpk solver.

.....some code..........
print("Objective "+objective_name)
with model:
    add_loopless(model)
    try:
        sol = model.optimize()
    except:
        print('model is infeasible')

print("loopless solution: status = " + sol.status)

......some other code..........

When I execute this,

Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_PSt3 "R_PSt3 Phosphatidylserine">
Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_PEt "R_PEt Phosphatidylethanolamine">
Objective max_nadph

In [1]:

Basically it did not print model is infeasible or the solution status. How can I guarantee a loopless solution with glpk solver? Any help appreciated.

**Comment by Midnighter** [comment]
If it doesn't print anything that means your program did not execute that line. I don't know how that might happen. This is also not the complete script so it's hard to tell what is going on. Please also note that if your objective is part of a loop, that loop cannot be broken.

**Comment by cdiener** [comment]
Agreed, we would need a [reproducible example](https://stackoverflow.com/help/mcve) to help out here. The Schellenberger method works for me:

```python
>>> from cobra.test import create_test_model
>>> from cobra.flux_analysis import add_loopless
>>> with mod:
...     add_loopless(mod)
...     sol = mod.optimize()
...
>>> sol.status
'optimal'
>>> sol.fluxes.head()
ACALD     0.00000
ACALDt    0.00000
ACKr      0.00000
ACONTa    6.00725
ACONTb    6.00725
Name: fluxes, dtype: float64
```

**Comment by sumanasri** [comment]
Thanks for your response. I will try debugging it with  a smaller example.

**Comment by sumanasri** [comment]
It does give an optimal solution. But in my case, the loop has not been removed. Will debug further. Please close this issue. I will reopen it if I have problems.
